### ISSUE TITLE
internal/cmd/gocdk: factor out Launcher into its own package

### ISSUE THREAD

**Issue by zombiezen** [google/go-cloud#2046]
The code for supporting `gocdk launch` is likely going to grow a bit beyond the support needed for other CLI subcommands. For code organization, it would be helpful to move the `Launcher` interface and its implementations into a separate package.

**Comment by zombiezen** [comment]
I've spent a little bit of time over last week thinking about this. In terms of code motion, I think the main goal is to move utility code away from command-line/argument-parsing code. As such, my plan is to:

1. Factor out all Docker subprocess logic together into a separate package. (This also moves us closer for #1804, should we choose to go that way.) There's a bunch of cross-cutting code (especially in launch) that would benefit from having a type-safe wrapper around the Docker CLI invocations, and would address the concern around passing `dockerEnv []string` variables around in a neater way.
2. Have one PR that is primarily focused at moving the relevant functions from launch.go into its own package. This should not change the API where possible (might need to export fields temporarily for visibility), but largely be a no-op code shuffle.
3. Have a final PR that makes API changes that we think are the right ones, now that the code is mostly in the right place.

@clausti, @vangent, WDYT?

**Comment by vangent** [comment]
SGTM. For, 3), I still think it's interesting to think about whether you could write an implementation of the `Launcher` API based on a plug-in CLI. I think that's mostly about make the arguments relatively simple to serialize.

**Comment by clausti** [comment]
SGTM, too.

**Comment by zombiezen** [comment]
> For, 3), I still think it's interesting to think about whether you could write an implementation of the `Launcher` API based on a plug-in CLI. I think that's mostly about make the arguments relatively simple to serialize.

Ah, I think this is actually the intuitive reason I structured `LaunchInput` the way I did: these are all the arguments to a launch (i.e. a Docker image reference, `launch_environment`, and `launch_specifier`) and come in through deserialization.

We could definitely add such a structure later, but it's unclear to me what benefit it would provide over a CLI that reads these inputs from files or `terraform output` itself (as that is the same contract we are holding ourselves to).
